third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1213,23 +1215,62 @@ struct ExpOpConversionApprox
                                    ConversionPatternRewriter &rewriter,
                                    Type elemTy, MultipleOperandsRange operands,
                                    Location loc) const {
-    // For non-FP32 input, call __nv_expf for higher-precision calculation
+    // For non-FP32 input, call __ocml_exp_f64 for higher-precision calculation
```

Comment:
So now for all cases we perserve denorm. Do we still need to defer to another pattern for the "higher-precision calculation" case? What about handle all cases directly here? Reason being avoid all the branches and different patterns causing confusion and potentially pitfalls for us.

Reply:
Unfortunately for higher precision exp function, we need `__ocml_exp_f64`, which is implemented by a bunch of llvm irs. Since we don't want to use `__ocml_xxx` here and we don't have a trivial llvm ir for exp_f64, I'd suggest we keep the lowering for f64 exp as is.

Reply:
wait, let me check what llvm.exp.f64 does

Reply:
It will fail to compile

Reply:
okay thanks for the explanation!